cpp/kiss_icp/3rdparty/eigen/eigen.cmake

```diff
@@ -32,16 +32,23 @@ set(EIGEN_BUILD_BLAS OFF CACHE BOOL "Don't build blas module")
 set(EIGEN_BUILD_LAPACK OFF CACHE BOOL "Don't build lapack module")
 
 include(FetchContent)
-FetchContent_Declare(eigen URL https://github.com/nachovizzo/eigen/archive/refs/tags/3.4.90.tar.gz)
-if(NOT eigen_POPULATED)
-  FetchContent_Populate(eigen)
-  if(${CMAKE_VERSION} GREATER_EQUAL 3.25)
-    add_subdirectory(${eigen_SOURCE_DIR} ${eigen_BINARY_DIR} SYSTEM EXCLUDE_FROM_ALL)
-  else()
-    # Emulate the SYSTEM flag introduced in CMake 3.25. Withouth this flag the compiler will
-    # consider this 3rdparty headers as source code and fail due the -Werror flag.
-    add_subdirectory(${eigen_SOURCE_DIR} ${eigen_BINARY_DIR} EXCLUDE_FROM_ALL)
-    get_target_property(eigen_include_dirs eigen INTERFACE_INCLUDE_DIRECTORIES)
-    set_target_properties(eigen PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${eigen_include_dirs}")
+if(${CMAKE_VERSION} GREATER_EQUAL 3.28)
+  FetchContent_Declare(eigen URL https://github.com/nachovizzo/eigen/archive/refs/tags/3.4.90.tar.gz EXCLUDE_FROM_ALL
+                                 SYSTEM)
+  FetchContent_MakeAvailable(eigen)
+else()
+  FetchContent_Declare(eigen URL https://github.com/nachovizzo/eigen/archive/refs/tags/3.4.90.tar.gz)
+  FetchContent_GetProperties(eigen)
```

Comment:
This is the real fix right? this should fix the issue without adding the extra 3.28., Or did I got it wrong?

Reply:
No, no, that will fix it as well; when you mention the guarding, I thought you wanted to keep the 3.28 version of things for future reference. I also misunderstood then, next time, phone call.

Reply:
Maybe worth leaving a comment for the future with a link to this PR, as in a few years maybe cmake 3.28 is mainstream and we can replace all these madness with those 2 lines